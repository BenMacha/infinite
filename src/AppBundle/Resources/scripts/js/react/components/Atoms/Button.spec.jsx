import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import Button from './Button.jsx';

describe('Button container', () => {
  it('render correctly', () => {
    const data = {
      text: 'The text to display',
    };
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };

    const tree = renderer.create(<Button store={store} {...data} />);

    // Test that onClick doest not throw error when fallback to default
    tree.toJSON().props.onClick();

    expect(tree).toMatchSnapshot();
  });

  it('default to empty text', () => {
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };
    const tree = renderer.create(<Button store={store} />);

    expect(tree).toMatchSnapshot();
  });

  it('dispatch onclick event', () => {
    const callback = jest.fn();
    const data = {
      onClick: callback,
      text: 'The text to display',
    };
    const store = {
      dispatch: jest.fn(),
      getState: jest.fn(),
      subscribe: jest.fn(),
    };

    const tree = renderer.create(<Button store={store} {...data} />).toJSON();

    tree.props.onClick();

    expect(callback).toBeCalled();
  });
});
