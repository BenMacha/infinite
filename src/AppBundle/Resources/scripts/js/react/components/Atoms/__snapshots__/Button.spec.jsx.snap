exports[`Button container default to empty text 1`] = `
<a
  className="ca-btn ca-btn--important"
  onClick={[Function]}>
  
</a>
`;

exports[`Button container render correctly 1`] = `
<a
  className="ca-btn ca-btn--important"
  onClick={[Function]}>
  The text to display
</a>
`;
